File: leetcode/Python/twosum.py
from typing import List

class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        cache = {}
        for i in range(len(nums)):
            comp = target - nums[i]
            if nums[i] in cache:
                return [cache[nums[i]], i]

            cache[comp] = i
        return []

sol = Solution()
nums = [2,7,11,15]
target = 9
print(sol.twoSum(nums, target))

File: leetcode/Lua/twosum.lua
local function two_sum(nums, target)
    local cache = {}
    for idx, val in ipairs(nums) do
        local comp = target - val
        if cache[val] then
            return { cache[val], idx }
        end
        cache[comp] = idx
    end
end

local nums = { 2, 7, 11, 15 }
local target = 9
print(table.concat(two_sum(nums, target), ", "))
